### PR TITLE
delay costly makeError to when it is needed

### PR DESCRIPTION
**Pull request by mgilbir**
In the context of issue #205, this change makes parsing 4x faster in my tests.

### CODE REVIEW

**Comment by vriad** [comment]
Can you summarize what you did here? Everything was prettified with a shorter `printWidth` so it's hard to find your changes.

**Comment by vriad** [comment]
Ah I just saw you're other comment. Basically your moving the `makeError` definition out of the parsing function. Makes sense! I'll play around with this.

**Comment by mgilbir** [comment]
Agh, sorry! I only saw now how messy the diff got due to applying prettier. I was rushing to send this between meetings and I didn't double check the diff before pushing 😞

I moved the `makeError` out fo the parsing because in the profiler I saw that the `ZodError` generation was always present and taking a significant amount of time on JSON blobs that never error. Doing less work on a heavy-hit function was bound to speed things up.

I didn't see any other obvious things in the profiler, so I didn't dig deeper.

**Comment by colinhacks** [comment]
Merged into Zod v1 and v2-beta. Thanks @mgilbir, great work on this!
